lib/models/account-manager.js

```diff
@@ -346,16 +346,32 @@ class AccountManager {
       email: userData.email,
       name: userData.name,
       externalWebId: userData.externalWebId,
-      webId: userData.webid || userData.webId ||
-        this.accountWebIdFor(userData.username)
+      localAccountId: userData.localAccountId,
+      webId: userData.webid || userData.webId || userData.externalWebId
     }
 
-    if (!userConfig.username) {
-      if (!userConfig.webId) {
+    try {
+      userConfig.webId = userConfig.webId || this.accountWebIdFor(userConfig.username)
+    } catch (err) {
+      if (err.message === 'Cannot construct uri for blank account name') {
         throw new Error('Username or web id is required')
+      } else {
+        throw err
       }
+    }
 
-      userConfig.username = this.usernameFromWebId(userConfig.webId)
+    if (userConfig.username) {
+      if (userConfig.externalWebId && !userConfig.localAccountId) {
+        // External Web ID exists, derive the local account id from username
+        userConfig.localAccountId = this.accountWebIdFor(userConfig.username)
```

Comment:
Sort of. the localAccountId just maps from the local server's username-based url to the external Web ID